types/materialize-css/index.d.ts

```diff
@@ -227,6 +227,128 @@ declare global {
             onDraw: (this: DatePicker) => void;
         }
 
+        interface DropdownOptions {
+			/**
+			 * Defines the edge the menu is aligned to
+			 * @default 'left'
+			 */
+            alignment: string;
+
+			/**
+			 * If true, automatically focus dropdown el for keyboard
+			 * @default true
+			 */
+            autoTrigger: boolean;
+
+			/**
+			 * If true, constrainWidth to the size of the dropdown activator
+			 * @default true
+			 */
+            constrainWidth: boolean;
+
+			/**
+			 * Provide an element that will be the bounding container of the dropdown
+			 * @default null
+			 */
+            container: Element;
+
+			/**
+			 * If false, the dropdown will show below the trigger
+			 * @default true
+			 */
+            coverTrigger: boolean;	// If false, the dropdown will show below the trigger.
+
+			/**
+			 * If true, close dropdown on item click
+			 * @default true
+			 */
+            closeOnClick: boolean;
+
+			/**
+			 * If true, the dropdown will open on hover
+			 * @default false
+			 */
+            hover: boolean;
+
+			/**
+			 * The duration of the transition enter in milliseconds
+			 * @default 150
+			 */
+            inDuration: number;
+
+			/**
+			 * The duration of the transition out in milliseconds
+			 * @default 250
+			 */
+            outDuration: number;
+
+			/**
+			 * Function called when dropdown starts entering
+			 * @default null
+			 */
+            onOpenStart: () => void;
```

Comment:
From Materialize source code, these callbacks are called with `this` set to the component itself and the first parameter set to the `Element`. https://github.com/Dogfalo/materialize/blob/v1-dev/js/dropdown.js#L549

Although this is not mentioned in the document, I think this is stable and can be useful sometimes